satellite/satellitedb/overlaycache.go

```diff
@@ -85,41 +103,91 @@ func (cache *overlaycache) SelectNewStorageNodes(ctx context.Context, count int,
 		args = append(args, v.Major, v.Major, v.Minor, v.Minor, v.Patch)
 	}
 
-	return cache.queryFilteredNodes(ctx, criteria.Excluded, count, safeQuery, args...)
+	var totalNodes []*pb.Node
+	for i := 0; i < 3; i++ {
```

Comment:
same question. In what case would calling `cache.queryFilteredNodes` a second time result in more results?

Reply:
I agree, thats not a good idea.